### DISCUSSION TITLE
Need help about "A2LParser-2.3.0-jar-with-dependencies.jar"

### DISCUSSION THREAD

**Discussion by ramkrganesh** [Luncher91/A2LParser#39] [General]
Hi,
I am trying to use the A2L parser but, I am unable to figure the part about "A2LParser-2.3.0-jar-with-dependencies.jar".
Short background : 
I use python scripting and would like to access the Java object structure returned by this package.
And here are few questions -
1) Is the "A2LParser-2.3.0-jar-with-dependencies.jar" be somehow created by me using the existing Java files from the repository ?
2) In the README.md file, there is a reference to "A2LParser.jar". Is this also a file to be generated by myself ?
Coming from C and Python background, maybe the questions I have asked are obvious for a java developer. but, please excuse me for my lack of understanding.
Thanks !
BR,
Ram

**Comment by Luncher91** [comment]
Hi Ram,

Sure, no worries!

The only jar file you would need is the "A2LParser-#-jar-with-dependencies.jar" (# replaces the version you are using; the most current one is 2.6.3).
Whenever I reference A2LParser.jar I actually mean the mentioned jar "with-dependencies".

You can compile it by yourself using maven. The easier way would be to use the already prepared versions attached to the [releases of the Repository ](https://github.com/Luncher91/A2LParser/releases)
There you can download the "with-dependencies" jar file and also the .py file which contains the generated classes for python.

I prepared a sample how to use the library with python in https://github.com/Luncher91/A2LParser/blob/master/codegen_python/sample.py
Summarized it is a matter of "import a2l", running the jar file with parameters to convert the A2L to Json and parsing the json then into the Object structure by calling "a2l.Asap2File.parse_raw(jsonContent)".

Does that help you to get started?

**Comment by ramkrganesh** [comment]
Hello Lenzen,

Thank you very much for the detailed explanation. The script works very well.
I was also able to parse an actual complex A2L from my project.

I will make further testings and will reach out to you in case I encounter an issue which I could not solve.

Thanks !
BR,
Ram
